lab-david/lib/fp.js

```diff
@@ -21,7 +21,7 @@ fp.filter = (callback, collection) => {
 };
 
 fp.slice = (begin, end, collection) => {
-  if(typeof callback !== 'function')
-    throw new TypeError('<callback> should be a function!');
-  return Array.prototype.slice.apply(this, [begin, end, collection]);
+  if(typeof callback !== 'object')
```

Comment:
did changing these lines and reordering where the collection is passed change anything?

Reply:
I changed it to `fp.slice = (begin, end, collection) => {
  if(!collection.length)
    throw new TypeError('<collection> is not an array-like object because it has no length!');
  return Array.prototype.slice.apply(this, [collection, begin, end]);
};` thinking I was checking for the wrong type but alas it throws an error. I dont think I understand where else I could pass the collection in.